### ISSUE TITLE
Support Kerberos auth for Changefeed Kafka Sink

### ISSUE THREAD

**Issue by chriscasano** [cockroachdb/cockroach#44603]
Customer request to have Kerberos auth for Kafka sink in Changefeed.  They have multiple scenarios for using changefeeds for audit events, table replication and other triggering events.  All production Kafka clusters us Kerberos and they can not migrate current app to production without this.  We would have to consider how we add Kerberos service name, credentials service, principal and key tab as part to the Create Changeed statement.  I haven’t tested but looks like Sarama supports Kerberos:  https://github.com/Shopify/sarama/pull/1366

**Comment by knz** [comment]
cc @nstewart for triage

**Comment by RoachietheSupportRoach** [comment]
Zendesk ticket #[4609](https://cockroachdb.zendesk.com/agent/tickets/4609) has been linked to this issue.

**Comment by BramGruneir** [comment]
@mwang1026 or @thtruo, any movement on this?

**Comment by knz** [comment]
Wasn't this achieved already?
@stevendanna @dt ?

**Comment by shermanCRL** [comment]
See https://github.com/Shopify/sarama/pull/1366

**Comment by knz** [comment]
@shermanCRL thanks, but how does this answer the question? Was the outcome from crdb's perspective achieved or not? In which version? Our CEAs would like to know.

**Comment by stevendanna** [comment]
No, we do not currently support this.  We added support for SCRAM as a new SASL authentication method but not GSSAPI.

**Comment by BramGruneir** [comment]
@amruss, any chance of resurrecting this?  I'm hearing from users who still really want this integration.
